### PR TITLE
First pass updating CONTRIBUTING

### PR DESCRIPTION
**Pull request by justincormack**
This largely removes references to Docker where possible.

The HOWTO guides are still Docker docs and refer to Docker not
moby, so the next step is to rework these as Moby docs and put
them in `docs/` in this repo.

See #35115

Signed-off-by: Justin Cormack <justin.cormack@docker.com>

talented-mutt-balancing

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -96,52 +92,28 @@ However, there might be a way to implement that feature *on top of* Docker.
...
-        <strong>irc.freenode.net</strong>.
-        IRC is a rich chat protocol but it can overwhelm new users. You can search
-        <a href="https://botbot.me/freenode/docker/#" target="_blank">our chat archives</a>.
-      </p>
-      <p>
-        Read our <a href="https://docs.docker.com/opensource/get-help/#irc-quickstart" target="_blank">IRC quickstart guide</a>
-        for an easy way to get started.
+        Register for the Docker Community Slack at
+	<a href="https://community.docker.com/registrations/groups/4316" target="_blank">https://community.docker.com/registrations/groups/4316</a>.
+	Archives are available at <a href="https://dockercommunity.slackarchive.io/" target="_blank">https://dockercommunity.slackarchive.io/</a>.
```
Comment (AkihiroSuda) [inline CONTRIBUTING.md]: Please mention #moby-project channel?

Comment (justincormack) [reply CONTRIBUTING.md]: added, thanks

**Review by vdemeester — APPROVED** [review]
LGTM 🐮

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
-# Contributing to Docker
+# Contributing to the Moby Project
```
Comment (mstanleyjones) [inline CONTRIBUTING.md]: Maybe this is a good opportunity to stop using the gerund form for titles "Contribute to the Moby Project" perhaps?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
 
-Want to hack on Docker? Awesome!  We have a contributor's guide that explains
-[setting up a Docker development environment and the contribution
+Want to hack on the Moby Project? Awesome! We have a contributor's guide that explains
+[setting up a development environment and the contribution
 process](https://docs.docker.com/opensource/project/who-written-for/).
```
Comment (mstanleyjones) [inline CONTRIBUTING.md]: Update this to live in the Moby repo now? Could just view it on Github directly.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -21,14 +21,14 @@ start participating.
 
 ## Reporting security issues
 
-The Docker maintainers take security seriously. If you discover a security
+The Moby maintainers take security seriously. If you discover a security
 issue, please bring it to their attention right away!
 
 Please **DO NOT** file a public issue, instead send your report privately to
 [security@docker.com](mailto:security@docker.com).
```
Comment (mstanleyjones) [inline CONTRIBUTING.md]: Should there be a non-Docker alias for this?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -83,11 +83,7 @@ contributions, see [the advanced contribution
 section](https://docs.docker.com/opensource/workflow/advanced-contributing/) in
 the contributors guide.
 
-We try hard to keep Docker lean and focused. Docker can't do everything for
-everybody. This means that we might decide against incorporating a new feature.
-However, there might be a way to implement that feature *on top of* Docker.
-
-### Talking to other Docker users and contributors
+### Talking to other Moby Project contributors
```
Comment (mstanleyjones) [inline CONTRIBUTING.md]: Maybe change to "Connect with..."

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -96,52 +92,29 @@ However, there might be a way to implement that feature *on top of* Docker.
...
-        IRC a direct line to our most knowledgeable Docker users; we have
-        both the  <code>#docker</code> and <code>#docker-dev</code> group on
-        <strong>irc.freenode.net</strong>.
-        IRC is a rich chat protocol but it can overwhelm new users. You can search
-        <a href="https://botbot.me/freenode/docker/#" target="_blank">our chat archives</a>.
-      </p>
-      <p>
-        Read our <a href="https://docs.docker.com/opensource/get-help/#irc-quickstart" target="_blank">IRC quickstart guide</a>
-        for an easy way to get started.
+        Register for the Docker Community Slack at
```
Comment (mstanleyjones) [inline CONTRIBUTING.md]: Do we want to run a Moby slack?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -266,15 +239,11 @@ Please see the [Coding Style](#coding-style) for further guidelines.
...
 ### Merge approval
 
-Docker maintainers use LGTM (Looks Good To Me) in comments on the code review to
-indicate acceptance.
-
-A change requires LGTMs from an absolute majority of the maintainers of each
-component affected. For example, if a change affects `docs/` and `registry/`, it
-needs an absolute majority from the maintainers of `docs/` AND, separately, an
-absolute majority of the maintainers of `registry/`.
+Moby maintainers use LGTM (Looks Good To Me) in comments on the code review to
```
Comment (mstanleyjones) [inline CONTRIBUTING.md]: Do we want to update this to use the Github built-in review process?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -266,15 +239,11 @@ Please see the [Coding Style](#coding-style) for further guidelines.
...
+indicate acceptance, or use the Github review approval feature.
 
-For more details, see the [MAINTAINERS](MAINTAINERS) page.
+For an explanation of the review and approval process see the
+[REVIEWING](project/REVIEWING.md) page.
 
 ### Sign your work
```
Comment (mstanleyjones) [inline CONTRIBUTING.md]: Is this still required? I believe this was a requirement of Docker legal. @thaJeztah PTAL

**Review by mdlinville — COMMENTED** [review]
Thanks for this. I left a few comments but I may be off base.

**Comment by londoncalling** [comment]
@justincormack thanks, LGTM, and I agree we should probably move this into https://github.com/moby/moby/. However, I talked with @johndmulhausen about this guide a while back, and we will probably want to have a shorter version also here on docker.github.io more specifically about contributing to the docs.

File: CONTRIBUTING.md (hunk 8 above)
Comment (thaJeztah) [reply CONTRIBUTING.md]: I think we should keep it for now, but we can keep the option open to remove the requirement at some point (the opposite is harder)

File: CONTRIBUTING.md (hunk 4 above)
Comment (justincormack) [reply CONTRIBUTING.md]: there will be shortly, it hadnt been set up yesterday, will check again today...

File: CONTRIBUTING.md (hunk 6 above)
Comment (justincormack) [reply CONTRIBUTING.md]: Eventually, yes, but telling everyone to move is disruptive so planning to keep using the Docker one for now.

**Comment by AkihiroSuda** [comment]
ping @justincormack

**Comment by londoncalling** [comment]
@justincormack LGTM, do you want me to merge this for you?

**Comment by justincormack** [comment]
I am just going to make some updates for this (sorry, was away).

**Comment by justincormack** [comment]
Ok, I ported the getting started contributing guide as well now. It will need a bit more tweaking as we change the binary names, and I had to adapt a bit to changes.

File: docs/contributing/test-and-docs.md [hunk 9]
```diff
@@ -0,0 +1,234 @@
+### Run tests and test documentation
```
Comment (thaJeztah) [inline docs/contributing/test-and-docs.md]: Can you remove the `and test documentation` here, and probably rename the file (and links to this file)?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
did a quick scan of images used/not used;

used:

- branch-sig.png
- contributor-edit.png
- contributors.png
- copy_url.png
- fork_docker.png
- git_bash.png
- list_example.png

not used:

- box.png
- checked.png
- fresh_container.png
- give_try.png
- include_gcc.png
- in_room.png
- path_variable.png
- red_notice.png
- three_running.png
- three_terms.png
- windows-env-vars.png
- windows-mingw.png

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Comment by thaJeztah** [comment]
CI is not testing this, and `dco-signed` is the only important one here; merging

**Pull request merged**